TweetMate/src/main/java/com/ybsystem/tweetmate/models/entities/twitter/TwitterStatus.java

```diff
@@ -0,0 +1,298 @@
+        MediaEntity[] medias = status.getMediaEntities();
+        HashtagEntity[] hashtags = status.getHashtagEntities();
+        UserMentionEntity[] userMentions = status.getUserMentionEntities();
+        this.urlEntities = new TwitterURLEntity[urls.length];
+        this.mediaEntities = new TwitterMediaEntity[medias.length];
+        this.hashtagEntities = new TwitterHashtagEntity[hashtags.length];
+        this.userMentionEntities = new TwitterUserMentionEntity[userMentions.length];
+        for (int i = 0; i < urls.length; i++) {
+            this.urlEntities[i] = new TwitterURLEntity(urls[i]);
+        }
+        for (int i = 0; i < medias.length; i++) {
+            this.mediaEntities[i] = new TwitterMediaEntity(medias[i]);
+        }
+        for (int i = 0; i < hashtags.length; i++) {
+            this.hashtagEntities[i] = new TwitterHashtagEntity(hashtags[i]);
+        }
+        for (int i = 0; i < userMentions.length; i++) {
+            this.userMentionEntities[i] = new TwitterUserMentionEntity(userMentions[i]);
+        }
+
+        this.createdAt = status.getCreatedAt();
+        this.id = status.getId();
+        this.quotedStatusId = status.getQuotedStatusId();
+        this.currentUserRetweetId = status.getCurrentUserRetweetId();
+
+        this.lang = status.getLang();
+        this.source = status.getSource();
+        this.text = status.getText();
+        this.displayTextRangeStart = status.getDisplayTextRangeStart();
+        this.displayTextRangeEnd = status.getDisplayTextRangeEnd();
+
+        // In reply to
+        this.inReplyToStatusId = status.getInReplyToStatusId();
+        this.inReplyToUserId = status.getInReplyToUserId();
+        this.inReplyToScreenName = status.getInReplyToScreenName();
+
+        // Count
+        this.favoriteCount = status.getFavoriteCount();
+        this.retweetCount = status.getRetweetCount();
+
+        // State
+        this.isFavorited = status.isFavorited();
+        this.isRetweeted = status.isRetweeted();
+        this.isRetweet = rtStatus != null;
+        this.isRetweetedByMe = currentUserRetweetId != -1L;
+
+        // Custom
+        this.convertedText = createTweetText(status);
+        this.convertedRelativeTime = createRelativeTime(status);
+        this.convertedAbsoluteTime = createAbsoluteTime(status);
+        this.convertedVia = createVia(status);
+        this.imageUrls = createImageUrls(status);
+        this.isMyTweet = status.getUser().getId() == TweetMateApp.getMyUser().getId();
+        this.isPublic = isMyTweet || !status.getUser().isProtected();
+        this.isTalk = status.getInReplyToStatusId() != -1;
+        this.isDetail = false;
+        this.isTwiMedia = (!imageUrls.isEmpty() && mediaEntities.length > 0);
+        this.isThirdMedia = (!imageUrls.isEmpty()  && mediaEntities.length == 0);
+        this.isVideo = (isTwiMedia && !mediaEntities[0].getType().equals("photo"));
+    }
+
+    private static String createTweetText(Status status) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)